src/compiler/checker.ts

```diff
@@ -3581,7 +3581,32 @@ namespace ts {
                         // An override clause will take effect for type-only imports and import types, and allows importing the types across formats, regardless of
                         // normal mode restrictions
                         if (isSyncImport && sourceFile.impliedNodeFormat === ModuleKind.ESNext && !getResolutionModeOverrideForClause(overrideClause)) {
-                            error(errorNode, Diagnostics.Module_0_cannot_be_imported_using_this_construct_The_specifier_only_resolves_to_an_ES_module_which_cannot_be_imported_synchronously_Use_dynamic_import_instead, moduleReference);
+                            if (findAncestor(location, isImportEqualsDeclaration)) {
+                                // ImportEquals in a ESM file resolving to another ESM file
+                                error(errorNode, Diagnostics.Module_0_cannot_be_imported_using_this_construct_The_specifier_only_resolves_to_an_ES_module_which_cannot_be_imported_with_require_Use_an_ECMAScript_import_instead, moduleReference);
+                            }
+                            else {
+                                // CJS file resolving to an ESM file

```

Comment:
This looks good but e37236d make it easier without having to get the scopes again. 
Also note that if you are merging this for 4.8 dont wait on the above change, as i am still working on it.
But you would want to pick 0eaf0f2 and d67505c since watch mode is not passing through module resolution cache which means in watch mode we would get this info from disk instead of cache.

Reply:
0eaf0f2 doesn't seem to (yet) refer to a linked commit.

Reply:
For some reason it doesnt get resolved like that 0eaf0f2122b261a47cbba801e88be7bc3b0f45c1